docs/guides/runbooks_steps_firehydrant.md

```diff
@@ -0,0 +1,779 @@
+allows all services related to a functionality to be automatically added to an incident, 
+giving you assurance that the right teams and services will be associated with your incident 
+no matter what.
+
+### Add Services Related to Functionality - Example Usage
+
+Basic usage:
+```hcl
+data "firehydrant_runbook_action" "firehydrant_add_services_related_to_functionality" {
+  integration_slug = "patchy"
+  slug             = "add_services_related_to_functionality"
+  type             = "incident"
+}
+
+resource "firehydrant_runbook" "firehydrant_add_services_related_to_functionality_runbook" {
+  name        = "add-services-related-to-functionality-runbook"
+  description = "This is an example configuration for a runbook that uses the FireHydrant add services related to functionality step"
+
+  steps {
+    name             = "Auto-Add Services Related To Functionality"
+    action_id        = data.firehydrant_runbook_action.firehydrant_add_services_related_to_functionality.id
+    automatic        = false
+    repeats          = true
+    repeats_duration = "PT15M"
+  }
+}
+```
+
+### Add Services Related to Functionality - Steps Argument Reference
+
+* `action_id` - (Required) The ID of the runbook action for the step.
+* `name` - (Required) The name of the step.
+* `automatic` - (Optional) Whether this step should automatically execute.
+* `repeats` - (Optional) Whether this step should repeat. Defaults to `false`.
+  When this value is `true`, `repeats_duration` _must_ be provided.
+* `repeats_duration` - (Optional) How often this step should repeat in ISO8601.
+  Example: PT10M [Format Spec](https://www.digi.com/resources/documentation/digidocs/90001437-13/reference/r_iso_8601_duration_format.htm)
+  This value _must_ be provided if `repeats` is `true`. This value _must not_ be provided if `repeats` is `false`.
+
+## Add Task List
+
+The FireHydrant **Add Task List** step allows 
+[FireHydrant task lists](https://support.firehydrant.com/hc/en-us/articles/5505273345428-Task-lists)
+to be automatically added to an incident.
+
+### Add Task List - Example Usage
+
+Basic usage:
+```hcl
+data "firehydrant_runbook_action" "firehydrant_add_task_list" {
+  integration_slug = "patchy"
+  slug             = "add_task_list"
+  type             = "incident"
+}
+
+resource "firehydrant_runbook" "firehydrant_add_task_list_runbook" {
+  name        = "add-task-list-runbook"
+  type        = "incident"
+  description = "This is an example configuration for a runbook that uses the FireHydrant add task list step"
+
+  steps {
+    name      = "Attach a Task List"
+    action_id = data.firehydrant_runbook_action.firehydrant_add_task_list.id
+
+    config = jsonencode({
+      task_list = {
+        label = "To Do List"
+        value = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+      }
+      role_for_assignment = {
+        label = "Commander"
+        value = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+      }
```

Comment:
It's confusing having label be something we input here since we shouldn't be deciding that here. We should just be using the ID. 

Shoot. I guess that's just how the API works...

Reply:
agreed! it is pretty confusing but you are right that that is how the API works. i initially wanted to implement this in a way that abstracted more of this away from the config for users but i realized that was going to take several weeks to get working :[

Reply:
TBD, what happens if the label doesn't match the role and or task list

Reply:
It appears to just overwrite the displayed name in the UI. The link still works just fine, so it's just like a name override for the runbook.